Azure Replication not working after 2.0 Upgrade

Hi,

after the Harbor upgrade to 2.0 the replication to our Azure Container Registry does not work anymore. The registry is marked as unhealthy.
When I try to add the registry again I get the error message "Failed to ping endpoint".
In the debug logs, I see that the acr responds to the ping request with unauthorized.
Does harbor use the given credentials for its registry ping request?

Hello,

We installed the 2.0.1 as soon as it was up, and sadly this problem is not solved.

We are able to push images from the command line, docker login on the same machine works flawlessly, but a simple ping in the UI will issue an error in the logs:

> Jun 22 09:16:13 172.27.0.1 core[31232]: 2020-06-22T09:16:13Z [ERROR] [/replication/adapter/native/adapter.go:126]: failed to ping registry https://xxx.azurecr.io: http status code: 401, body: {"errors":[{"code":"UNAUTHORIZED","message":"authentication required, visit https://aka.ms/acr/authorization for more information.","detail":null}]}

Note that with harbor version 1 there is no issue at all.

--- a/src/pkg/registry/auth/authorizer.go
+++ b/src/pkg/registry/auth/authorizer.go
@@ -54,9 +54,9 @@ type authorizer struct {
 }
 
 func (a *authorizer) Modify(req *http.Request) error {
-	// Nil URL means this is the first time the authorizer is called
+	// Nil underlying authorizer means this is the first time the authorizer is called
 	// Try to connect to the registry and determine the auth scheme
-	if a.url == nil {
+	if a.authorizer == nil {
 		// to avoid concurrent issue
 		a.Lock()
 		defer a.Unlock()
@@ -74,7 +74,7 @@ func (a *authorizer) Modify(req *http.Request) error {
 }
 
 func (a *authorizer) initialize(u *url.URL) error {
-	if a.url != nil {
+	if a.authorizer != nil {
 		return nil
 	}
 	url, err := url.Parse(u.Scheme + "://" + u.Host + "/v2/")

--- a/src/pkg/registry/auth/bearer/authorizer.go
+++ b/src/pkg/registry/auth/bearer/authorizer.go
@@ -142,5 +142,8 @@ func (a *authorizer) fetchToken(scopes []*scope) (*token, error) {
 	if err = json.Unmarshal(body, token); err != nil {
 		return nil, err
 	}
+	if len(token.Token) == 0 && len(token.AccessToken) > 0 {
+		token.Token = token.AccessToken
+	}
 	return token, nil
 }

